internal/controller/isbservicerollout_controller.go

```diff
@@ -300,13 +310,19 @@ func (r *ISBServiceRolloutReconciler) processExistingISBService(ctx context.Cont
 			r.customMetrics.ReconciliationDuration.WithLabelValues(ControllerISBSVCRollout, "update").Observe(time.Since(syncStartTime).Seconds())
 			return nil
 		})
-	} else if isbServiceNeedsUpdating {
-		// update ISBService
-		err = r.updateISBService(ctx, isbServiceRollout, newISBServiceDef)
-		if err != nil {
-			return false, err
+	case config.NoStrategyID:
```

Comment:
Hmm, we also needs to track for ISBService if there is an ongoing upgrade happening now, right?

Reply:
yes. But in the current code, we are doing PPND for all updates if that's the user strategy for the namespace. Once we incorporate the logic that looks at the individual fields for isbsvc to do apply vs PPND/Progressive, then we need InProgressStrategy here too.

Reply:
I see - maybe in this case you are also thinking about when the user changes from one strategy to another in their configmap? Is it okay if we include this as part of the work Dillen will do for isbsvc?

Reply:
I have updated the description in that [issue](https://github.com/numaproj/numaplane/issues/248) to make sure this is included. I will also add a comment for that issue for what the pseudocode should be.

Reply:
sounds good to me. Thanks!